prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -1994,6 +1997,197 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                             );
                         }
 
+                        "prusti_contracts::ghost::GhostInt::new" => {
+                            // let sample_ghost_int = GhostInt::new(ghost_val);
+                            //args[0] will have the integer value `ghost_val`
+                             
+                            // N.B. Do we need to encode the destination for ghost variables as well?
+                            let &(ref target_place, _) = destination.as_ref().unwrap();
+                            let (dst, dest_ty, _) = self.mir_encoder.encode_place(target_place).unwrap();
```

Comment:
It would be better to return the encoding error instead of calling `unwrap()`. We still have to finish doing these replacements in the codebase.